### DISCUSSION TITLE
USR W610

### DISCUSSION THREAD

**Discussion by wymes** [wills106/homeassistant-solax-modbus#461] [General]
Hi

I am experimenting with the above device . I know the actual com port is working  by testing with a laptop  through the device to a virtual com port over ether net . I can see no activity on the RS585 port when connected to the Solis inverter . The wiring is correct .and I can see approx 2.6v across the leads 

I have tried a 485 to 232 adapter  with no luck . I am awaiting an adapter to test as per the laptop test 

Any clues on the config 

THe unit is set up for tcp  mmodbus rtu  and the 485 port is enabled and the baud speed is correct

**Comment by alienatedsec** [comment]
Hi @wymes

Any screenshots of your configuration, and settings could be useful to oversee some of the challenges you are facing with this device.

**Comment by wymes** [comment]
Hi

 

As requested

**Reply by alienatedsec** [reply]
Sorry @wymes there is nothing in your response. It looks like responding via email removes screenshots.

**Comment by wymes** [comment]
Hi

Just to add . I am assuming that the solis inverter  is acting as a client to the w610 unit  or is it a server to the w610 unit  ?

REgards

Nigel

**Reply by alienatedsec** [reply]
@wymes If you are asking about Server / Client config for your W610, it will be in relation to your device. Server means that HA (or another Client) can query it, and the Client will be to connect to other Servers.

The inverter is a Slave on the Modbus network, which has nothing to do with Server / Client config.
However, your W610 wired to the inverter should be set as Server.

**Comment by wymes** [comment]
![106-2](https://github.com/wills106/homeassistant-solax-modbus/assets/70014350/16e288b4-7932-484e-8f6f-0ca46a0a4759)
![106-1](https://github.com/wills106/homeassistant-solax-modbus/assets/70014350/bf797388-13ec-44a5-af2d-7c3d1bfa1bc7)

**Reply by alienatedsec** [reply]
@wymes I am not sure what is the 485 selector switch mode. 

Apart from that, it looks like it uses an 8899 port. When configuring SolaX integration, did you use that port and IP 10.10.100.100?

Is that even the correct subnet on your network?

**Reply by wymes** [reply]
Hi

Apologies for splattering the discussions . What i have working is the w610 is connected via wifi to my router . It has an IP address from the router dhcp pool (192.168.0.x) . I use this address to connect the integration over port 8899 

If I use the lan ports for the second device they have a 10.x.x.x address 

From the entities i see you have the following 
Battery Input Energy  - charge 
Battery Output Energy  - Discharge 

Battery Power  one of the above 

Would be great if battery Power  was a -negative  value for discharge and a positive value for charge . Then you can have a needle gauge cantered on zero  which can then swing positive or negative 

The solis cloud integration shows the value as neg or pos :-) 

Ma\y thanks 

Nigel

**Reply by alienatedsec** [reply]
@wymes create this template replace `solislan` with your name

```
sensor:
  - platform: template
    sensors:
      battery_power_display:
        friendly_name: "Battery Power Display"
        unit_of_measurement: "W"
        value_template: >
          {% if states('sensor.solislan_battery_input_energy')|float > 0 %}
            +{{ states('sensor.solislan_battery_input_energy') }}
          {% elif states('sensor.solislan_battery_output_energy')|float > 0 %}
            -{{ states('sensor.solislan_battery_output_energy') }}
          {% else %}
            0
          {% endif %}
        icon_template: >
          {% if states('sensor.solislan_battery_input_energy')|float > 0 %}
            mdi:battery-plus
          {% elif states('sensor.solislan_battery_output_energy')|float > 0 %}
            mdi:battery-minus
          {% else %}
            mdi:battery-unknown
          {% endif %}
```

**Reply by alienatedsec** [reply]
@wymes 

Thanks for inspiring me to do the same as I wanted it for a while.

![image](https://github.com/wills106/homeassistant-solax-modbus/assets/73167064/cbd56a75-9b59-494a-9521-66e87f2f920a)

**Reply by wymes** [reply]
That worked a treat thanks 

![image](https://github.com/wills106/homeassistant-solax-modbus/assets/70014350/3005a6dd-c675-49b4-bf01-d93914c6aa0e)

**Comment by wymes** [comment]
Hi

Tried station mode with data transfer as transparent  and the integration burst into life :-) 

If I were to buy a second device not sure how that would connect to the first  so that the original data logger would work as well ?

Thanks 

Nigel

**Reply by alienatedsec** [reply]
@wymes 
It's hard to follow the conversation as you don't reply to the same thread but you make separate comments which create many threads in the same conversation. 

When you purchase another device, you wire it to a data logger and you have to configure it as a client and provide the IP of the server to connect to (the one that is currently working with the integration)

**Comment by wymes** [comment]
hopefully final question 

in the solis config I can see the following entity 

SolisModbusSensorEntityDescription(
        name = "Battery Power",
        key = "battery_power",
        native_unit_of_measurement = UnitOfPower.WATT,
        device_class = SensorDeviceClass.POWER,
        state_class = SensorStateClass.MEASUREMENT,
        register = 33149,
        register_type = REG_INPUT,
        unit = REGISTER_S32,
        entity_registry_enabled_default = False,
        allowedtypes = HYBRID,
        icon = "mdi:home",

But in the integration it is not showing  . Everything else appears to be there . 143 entities found ?


Regards

Nigel

**Reply by wymes** [reply]
looking at the the HA config  looks like it is disabled by the integration . I tried setting the value to null but that screwed up HA :-(
I will rebuild the server tomorrow with a fresh start 

Regards

Nigel

**Reply by alienatedsec** [reply]
@wymes it's not enabled by default. You can enable it in entities when you filter disabled ones.

**Comment by wymes** [comment]
Hi

The sensor you created above has been working fine . I upgraded HA core to 2026.6.0 today  and the on completion of the update the sensor  no longer works . I get the following errors 

Logger: homeassistant.helpers.entity_component
Source: helpers/entity_component.py:55
First occurred: 09:46:30 (950 occurrences)
Last logged: 12:24:30

Forced update failed. Entity sensor.battery_power_display not found.


<img width="575" height="448" alt="ha entity fail" src="https://github.com/user-attachments/assets/f08bed97-c923-42b9-a2d4-2caaa446fddf" />
<img width="462" height="119" alt="yaml warning" src="https://github.com/user-attachments/assets/51425486-c965-4b88-89db-edd3cbc81108" />


on the dashboard the entity  displays as not found 

Regards


Nigel

**Reply by wymes** [reply]
Found the resolution in the change notes for the release . Clear instructions on how to migrate legacy templates Phew
